CMSgov/mdct-mcr
# MDCT-1995: Add Delete Modal to DynamicFields

## Description
Fixes [MDCT-1995](https://qmacbis.atlassian.net/browse/MDCT-1995)

Also of note, this makes it so dynamic fields can't be deleted so there are no more fields on the screen. The last field will simply be cleaned of input and be ready for new input again.


https://user-images.githubusercontent.com/19439679/193655898-97de3978-26be-4e29-9b6f-28df852476b4.mov



### How to test
./dev local
Create a program
Go to the Add a Plan page or the BSS Entities page
Add Plans/BSS Entities
Delete them
Save the page and comeback to make sure they are all still there
Delete another/add another
Save
Come back and confirm changes

### Changed Dependencies


## Code author checklist

## Reviewer checklist (two different people)
